taipy/_entrypoint.py

```diff
@@ -71,3 +76,58 @@ def _entrypoint():
 
     _TaipyParser._remove_argument("help")
     _TaipyParser._parser.print_help()
+
+
+import argparse
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Taipy server")
+    # REST server arguments
+    parser.add_argument("--rest-port", type=int, help="Port to run the REST server on")
+    parser.add_argument("--rest-host", type=str, help="Host to run the REST server on")
+    parser.add_argument("--rest-use-https", action="store_true", help="Use HTTPS for REST server")
+    parser.add_argument("--rest-ssl-cert", type=str, help="Path to SSL certificate for REST server")
+    parser.add_argument("--rest-ssl-key", type=str, help="Path to SSL key for REST server")
+    # GUI server arguments
+    parser.add_argument("--gui-port", type=int, help="Port to run the GUI server on")
+    parser.add_argument("--gui-host", type=str, help="Host to run the GUI server on")
+    parser.add_argument("--gui-use-https", action="store_true", help="Use HTTPS for GUI server")
+    parser.add_argument("--gui-ssl-cert", type=str, help="Path to SSL certificate for GUI server")
+    parser.add_argument("--gui-ssl-key", type=str, help="Path to SSL key for GUI server")
+    return parser.parse_args()
+
+def main():
+    args = parse_args()
+
+    # Configure REST
+    rest_config = {
+        "port": args.rest_port or 5000,
+        "host": args.rest_host or "127.0.0.1",
+        "use_https": args.rest_use_https,
+        "ssl_cert": args.rest_ssl_cert,
+        "ssl_key": args.rest_ssl_key
+    }
+    rest = Rest(**rest_config)
+
+    # Configure GUI
+    gui_config = {
+        "port": args.gui_port or 5001,  # Note: different default port
+        "host": args.gui_host or "127.0.0.1",
+        "use_reloader": False,
+        "use_https": args.gui_use_https,
+        "ssl_certfile": args.gui_ssl_cert,
+        "ssl_keyfile": args.gui_ssl_key
+    }
+    gui = Gui()
+
+    # Run the application with both GUI and REST
+    gui.run(
+        run_server=True,
+        rest=rest,
+        **gui_config
+    )
+
+if __name__ == "__main__":
+    main()
```

Comment:
The `_entrypoint.py` file acts as the entrypoint of the application when you run `taipy run ..`

The Rest argument should not be handled here, but in the `Rest()` object when you run the Rest service, as this `main()` function will never be called